File: Need/slide/tests/unit/client_test.py
#!/usr/bin/env python3
"""
Client test for Socket.IO presentation generation.
"""

import socketio
import asyncio
import requests
import time

# Configuration
SERVER_URL = "http://127.0.0.1:8060"
TOKEN = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2ODgwODdlMzA1MTk0OTc2YWVhOTk0MDMiLCJzdWIiOiI2ODgwODdlMzA1MTk0OTc2YWVhOTk0MDMiLCJlbWFpbCI6InJycmlhZHVkZGluQGdtYWlsLmNvbSIsInBhY2thZ2UiOiJ1bmxpbWl0ZWQiLCJpc192ZXJpZmllZCI6dHJ1ZSwicm9sZSI6InVzZXIiLCJpYXQiOjE3NjEzNzczMzAsImV4cCI6MTc2MTQ2MzczMH0.dUeYJu4jNbaSfN8jRloiFiRTie1WkJ1prWtMe-_RQLg"
MESSAGE = "Create a presentation about AI in Healthcare"

# Socket.IO client
sio = socketio.AsyncClient()

@sio.event
async def connect():
    print("✅ Connected to Socket.IO")

@sio.event
async def disconnect():
    print("❌ Disconnected from Socket.IO")

@sio.event
async def agent_output(data):
    print(f"📨 Agent Output: {data}")

@sio.event
async def message(data):
    print(f"📨 Message: {data}")

async def create_presentation():
    response = requests.post(f"{SERVER_URL}/create-presentation", 
                           json={"message": MESSAGE, "file_urls": []},
                           headers={"Authorization": f"Bearer {TOKEN}"})
    return response.json()["p_id"] if response.status_code == 200 else None

async def start_presentation(p_id):
    response = requests.post(f"{SERVER_URL}/start-presentation/{p_id}",
                           headers={"Authorization": f"Bearer {TOKEN}"})
    return response.status_code == 200

async def main():
    print("🧪 Client Test Starting...")
    
    # Create presentation
    p_id = await create_presentation()
    print(f"📝 Created presentation: {p_id}")
    
    # Connect Socket.IO
    await sio.connect(f"{SERVER_URL}?p_id={p_id}&token={TOKEN}")
    print("🔌 Connected to Socket.IO")
    
    # Start presentation
    await start_presentation(p_id)
    print("🚀 Started presentation")
    
    # Wait for events
    await asyncio.sleep(30)
    
    # Disconnect
    await sio.disconnect()
    print("✅ Test completed")

if __name__ == "__main__":
    asyncio.run(main())

File: Need/slide/tests/client/client_test.js
const io = require('socket.io-client');

const SERVER_URL = 'http://127.0.0.1:8060';
const TOKEN = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2ODgwODdlMzA1MTk0OTc2YWVhOTk0MDMiLCJzdWIiOiI2ODgwODdlMzA1MTk0OTc2YWVhOTk0MDMiLCJlbWFpbCI6InJycmlhZHVkZGluQGdtYWlsLmNvbSIsInBhY2thZ2UiOiJ1bmxpbWl0ZWQiLCJpc192ZXJpZmllZCI6dHJ1ZSwicm9sZSI6InVzZXIiLCJpYXQiOjE3NjEzNzczMzAsImV4cCI6MTc2MTQ2MzczMH0.dUeYJu4jNbaSfN8jRloiFiRTie1WkJ1prWtMe-_RQLg';
const MESSAGE = 'Create a presentation about AI in Healthcare';

async function createPresentation() {
    const response = await fetch(`${SERVER_URL}/create-presentation`, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${TOKEN}`
        },
        body: JSON.stringify({
            message: MESSAGE,
            file_urls: []
        })
    });
    const data = await response.json();
    return data.p_id;
}

async function startPresentation(p_id) {
    const response = await fetch(`${SERVER_URL}/start-presentation/${p_id}`, {
        method: 'POST',
        headers: {
            'Authorization': `Bearer ${TOKEN}`
        }
    });
    return response.ok;
}

async function main() {
    console.log('🧪 Client Test Starting...');
    
    // Create presentation
    const p_id = await createPresentation();
    console.log(`📝 Created presentation: ${p_id}`);
    
    // Connect Socket.IO
    const socket = io(SERVER_URL, {
        query: {
            p_id: p_id,
            token: TOKEN
        }
    });
    
    socket.on('connect', () => {
        console.log('✅ Connected to Socket.IO');
    });
    
    socket.on('agent_output', (data) => {
        console.log('📨 Agent Output:', data);
    });
    
    socket.on('message', (data) => {
        console.log('📨 Message:', data);
    });
    
    // Start presentation
    await startPresentation(p_id);
    console.log('🚀 Started presentation');
    
    // Wait for events
    await new Promise(resolve => setTimeout(resolve, 30000));
    
    // Disconnect
    socket.disconnect();
    console.log('✅ Test completed');
}

main().catch(console.error);
